samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,749 @@
+ * also publish its status on a state change.
+ *
+ * Messages from a button to its corresponding LED are ignored as
+ * the LED's state has already been changed locally by the button client.
+ *
+ * The buttons are debounced at a nominal 250ms. That value can be
+ * changed as needed.
+ *
+ */
+
+/*
+ * Open Issues
+ *
+ * 1. Should the level skeleton remain or just be noted as still just
+ *    a skeleton?
+ *
+ */
+
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <nrf.h>
+#include <device.h>
+#include <gpio.h>
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/l2cap.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/mesh.h>
+#include <../../../../../subsys/bluetooth/host/hci_core.h>
+#include <../../../../../subsys/bluetooth/common/log.h>
+
+#include <board.h>
+
+#define CID_INTEL 0x0002
+
+/* Model ID's */
+#define BT_MESH_MODEL_ID_GEN_ONOFF_SRV             0x1000
+#define BT_MESH_MODEL_ID_GEN_ONOFF_CLI             0x1001
+
+/* Model Operations */
+#define BT_MESH_MODEL_OP_GEN_ONOFF_GET		BT_MESH_MODEL_OP_2(0x82, 0x01)
+#define BT_MESH_MODEL_OP_GEN_ONOFF_SET		BT_MESH_MODEL_OP_2(0x82, 0x02)
+#define BT_MESH_MODEL_OP_GEN_ONOFF_SET_UNACK	BT_MESH_MODEL_OP_2(0x82, 0x03)
+#define BT_MESH_MODEL_OP_GEN_ONOFF_STATUS	BT_MESH_MODEL_OP_2(0x82, 0x04)
+
+/*
+ * Server Configuration Declaration
+ */
+
+static struct bt_mesh_cfg_srv cfg_srv = {
+	.relay = BT_MESH_RELAY_DISABLED,
+	.beacon = BT_MESH_BEACON_ENABLED,
+#if defined(CONFIG_BT_MESH_FRIEND)
+	.frnd = BT_MESH_FRIEND_ENABLED,
+#else
+	.frnd = BT_MESH_FRIEND_NOT_SUPPORTED,
+#endif
+#if defined(CONFIG_BT_MESH_GATT_PROXY)
+	.gatt_proxy = BT_MESH_GATT_PROXY_ENABLED,
+#else
+	.gatt_proxy = BT_MESH_GATT_PROXY_NOT_SUPPORTED,
+#endif
+	.default_ttl = 7,
+
+	/* 3 transmissions with 20ms interval */
+	.net_transmit = BT_MESH_TRANSMIT(2, 20),
+	.relay_retransmit = BT_MESH_TRANSMIT(2, 20),
+};
+
+/*
+ * Client Configuration Declaration
+ */
+
+static struct bt_mesh_cfg_cli cfg_cli = {
+};
+
+/*
+ * Health Client/Server Declarations
+ */
+
+static struct bt_mesh_health_srv health_srv = {
+};
+
+static struct bt_mesh_health_srv health_cli = {
+};
+
+/*
+ * Publication Declarations
+ *
+ * The publication messages are initialized to the
+ * the size of the opcode + content + MIC
+ *
+ * The messages are in static storage because NET_BUF_SIMPLE()
+ * only allocates on the stack if called within a function.
+ * For publication, the message must be in static or global as
+ * it is re-transmitted several times. This occurs
+ * after the function that called bt_mesh_model_publish() has
+ * exited and the stack is no longer valid.
+ *
+ */
+
+static struct bt_mesh_model_pub health_pub = {
+	.msg  = BT_MESH_HEALTH_FAULT_MSG(0),
+};
+
+static struct bt_mesh_model_pub gen_level_pub;
+static struct bt_mesh_model_pub gen_onoff_pub_srv = {
+	.msg = NET_BUF_SIMPLE(2 + 2 + 4),
```

Comment:
I missed this the first time around: I suppose the "+ 4" is intended for the AppMIC? That's not needed for the publication message, since that one is the "pristine" message which gets copied to a stack variable (with space for the MIC) each time a transmission occurs (the design is like this due to the Publish Retransmit state).

Reply:
Fixed